### PR TITLE
Distribute functional tests

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the Windows code?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Related PRs or issues

[AB#74714](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/74714)

### CODE REVIEW

File: Tests/Contoso.Test.Functional.Droid/MainActivity.cs [hunk 1]
```diff
@@ -39,6 +44,8 @@ protected override void OnCreate(Bundle bundle)
             TerminateAfterExecution = true;
 #endif
 
+
+            AppCenter.Start(_appSecret, typeof(Distribute));
             // you cannot add more assemblies once calling base
```
Comment (MatkovIvan) [inline Tests/Contoso.Test.Functional.Droid/MainActivity.cs]: Format empty lines

File: Tests/Microsoft.AppCenter.Test.Functional/Distribute/DistributeUpdateTest.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            DependencyConfiguration.HttpNetworkAdapter = httpNetworkAdapter;
+
+            // Reset instance.
+            // TODO add UnsetInstance to Distribute.
+            // Distribute.UnsetInstance();
+            AppCenter.UnsetInstance();
+            
+            // Start App Center.
+            AppCenter.LogLevel = LogLevel.Verbose;
+            //AppCenter.Start(_appSecret, typeof(Distribute));
```
Comment (MatkovIvan) [inline Tests/Microsoft.AppCenter.Test.Functional/Distribute/DistributeUpdateTest.cs]: Remove or uncomment

File: Tests/Microsoft.AppCenter.Test.Functional/Config.cs [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
 
+using System;
+
 namespace Microsoft.AppCenter.Test.Functional
 {
     public static class Config
     {
         public const int ResultChannelPort = 16384;
+
+        public const string _appSecret = "e94aaff4-e80d-4fee-9a5f-a84eb6e688fc";
```
Comment (guperrot) [inline Tests/Microsoft.AppCenter.Test.Functional/Config.cs]: Constants are PascalCase

File: Tests/Microsoft.AppCenter.Test.Functional/Distribute/DistributeUpdateTest.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
+﻿using System.Threading.Tasks;
```
Comment (guperrot) [inline Tests/Microsoft.AppCenter.Test.Functional/Distribute/DistributeUpdateTest.cs]: copyright missing

File: Tests/Microsoft.AppCenter.Test.Functional/Distribute/DistributeUpdateTest.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            Distribute.UnsetInstance();
+            AppCenter.LogLevel = LogLevel.Verbose;
+            AppCenter.Start(Config._appSecret, typeof(Distribute));
+            DistributeEvent?.Invoke(this, DistributeTestType.OnResumeActivity);
+
+            // Wait when Distribute will start.
+            await Distribute.IsEnabledAsync();
+
+            // Disable and enable distribute.
+            Distribute.SetEnabledAsync(false).Wait();
```
Comment (guperrot) [inline Tests/Microsoft.AppCenter.Test.Functional/Distribute/DistributeUpdateTest.cs]: We should have the golden path not doing this, so maybe we need 2 tests.

**Comment by MatkovIvan** [comment]
CI should be fixed by #1298
Merging this as is to feature branch, future comments (if any) will be addressed separately

File: Tests/Microsoft.AppCenter.Test.Functional/Microsoft.AppCenter.Test.Functional.csproj [hunk 6]
```diff
@@ -7,12 +7,21 @@
...
     <ProjectReference Include="..\..\SDK\AppCenterAnalytics\Microsoft.AppCenter.Analytics\Microsoft.AppCenter.Analytics.csproj" />
+    <ProjectReference Include="..\..\SDK\AppCenterDistribute\Microsoft.AppCenter.Distribute\Microsoft.AppCenter.Distribute.csproj" />
   </ItemGroup>
   <ItemGroup>
     <Folder Include="Analytics\" />
+    <Folder Include="Distribute\" />
+  </ItemGroup>
+  <ItemGroup>
+    <Reference Include="Mono.Android">
+      <HintPath>..\..\..\..\..\..\Library\Frameworks\Xamarin.Android.framework\Versions\10.0.6.2\lib\xamarin.android\xbuild-frameworks\MonoAndroid\v8.0\Mono.Android.dll</HintPath>
```
Comment (MatkovIvan) [inline Tests/Microsoft.AppCenter.Test.Functional/Microsoft.AppCenter.Test.Functional.csproj]: It doesn't look correct

File: SDK/AppCenterDistribute/Microsoft.AppCenter.Distribute.iOS/Distribute.cs [hunk 7]
```diff
@@ -100,6 +100,11 @@ static void HandleUpdateAction(UpdateAction updateAction)
             }
         }
 
+        public static void UnsetInstance()
+        {
+            //MSDistribute.ResetSharedInstance();
```
Comment (MatkovIvan) [inline SDK/AppCenterDistribute/Microsoft.AppCenter.Distribute.iOS/Distribute.cs]: Uncomment or remove

File: SDK/AppCenterDistribute/Microsoft.AppCenter.Distribute.Android/Distribute.cs [hunk 8]
```diff
@@ -102,5 +102,10 @@ public bool OnReleaseAvailable(Activity activity, AndroidReleaseDetails androidR
                 return false;
             }
         }
+
+        public static void UnsetInstance()
+        {
+            //AndroidDistribute.UnsetInstance();
```
Comment (MatkovIvan) [inline SDK/AppCenterDistribute/Microsoft.AppCenter.Distribute.Android/Distribute.cs]: Uncomment or remove

**Pull request merged**